apps/api-extractor/src/api/ExtractorConfig.ts

```diff
@@ -292,7 +316,19 @@ export class ExtractorConfig {
    * its format may be changed at any time.
    */
   public getDiagnosticDump(): string {
-    const result: object = MessageRouter.buildJsonDumpObject(this);
+    // Handle the simple JSON-serializable properties using buildJsonDumpObject()
+    const result: object = MessageRouter.buildJsonDumpObject(this, {
+      keyNamesToOmit: ['tsdocConfigFile', 'tsdocConfiguration']
+    });
+
+    // Implement custom formatting for tsdocConfigFile and tsdocConfiguration
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (result as any).tsdocConfigFile = {
```

Comment:
Can we avoid the `as any`?

Reply:
With TSLint we used to be able to solve this problem by writing `result["tsdocConfigFile"] =` which basically  means I am treating `result` as a bag of key/value pairs, not as a typed object.  Next time I work on the lint rules, I'll try to get that fixed.